pkg/yum/yum.go

```diff
@@ -78,32 +215,26 @@ func GetPrimaryURL(repomd Repomd, url string) (string, error) {
 	if primaryLocation == "" {
 		return "", fmt.Errorf("GET error: Unable to parse 'primary' location in repomd.xml")
 	}
-
-	return fmt.Sprintf("%s/%s", url, primaryLocation), nil
+	return fmt.Sprintf("%s/%s", r.URL, primaryLocation), nil
 }
 
-// GetRepomdXML returns Repomd struct given http client and URL to yum repository
-func GetRepomdXML(client http.Client, url string) (Repomd, error) {
-	var result Repomd
+// fetchFile GET request url with timeout duration timeout. Returns response and error.
+func fetchWithTimeout(url string, timeout time.Duration) (*http.Response, error) {
+	transport := http.Transport{ResponseHeaderTimeout: timeout}
+	client := http.Client{Transport: &transport, Timeout: timeout}
+	return client.Get(url)
```

Comment:
wouldn't you want to use the client passed in by the caller to the repository struct?